src/app/app.js

```diff
@@ -1,32 +1,58 @@
 import express from 'express';
+import cookieSession from 'cookie-session';
 import helmet from 'helmet';
 import pinoMiddleware from 'express-pino-logger';
 import compression from 'compression';
 import staticGzip from 'express-static-gzip';
+import fetch from 'node-fetch';
+
+import auth from '../auth';
 import home from '../home';
 import orgs from '../orgs';
 import {pageNotFoundMiddleware, internalServerErrorMiddleware} from '../errors';
 import csp from './app.csp';
 
-export default function ({logger} = {}) {
+export default function (config) {
   const app = express();
 
-  if (logger) {
-    app.use(pinoMiddleware({logger}));
-  }
+  app.use(pinoMiddleware(config.logger));
+
+  app.use(cookieSession({
+    name: 'pazmin-session',
+    keys: [config.sessionSecret],
+    secure: config.isProduction,
+    httpOnly: true
+  }));
 
   app.use('/assets', staticGzip('dist/assets', {immutable: true}));
   app.use(compression());
 
   app.use(helmet());
   app.use(helmet.contentSecurityPolicy(csp));
 
-  app.use('/orgs', orgs);
   app.use('/', home);
 
+  // Authenticated endpoints follow
+  app.use(auth(config));
+
+  app.use('/orgs', orgs);
+
+  app.get('/tmp-orgs', (req, res) => {
```

Comment:
Should we be removing this before merging? ... I know it's temporary junk but could we not replace the hard coded url with a `${CF_API_ENDPOINT}/v2/organisations` or something and leave it as an example?

Reply:
Initially I had it in orgs an was keen to keep as an example for future stories. I believe @keymon had some opinions as to why we shouldn't do that.

Reply:
It's not massively important ... but in one glace at that function I worked out:

"oh so the auth stuff must give me a nice `req.session.passport.user` thing that probably contains the token"

if it goes away we should probably at least say something in the README like "our authentication middleware adds the `req.session.passport.user` property which contains an access_token